include/sparrow/array/array_data_factory.hpp

```diff
@@ -140,30 +182,37 @@ namespace sparrow
         {
             SPARROW_ASSERT_TRUE(check_all_elements_have_same_size(values));
         }
-
-        SPARROW_ASSERT_TRUE(values.size() == bitmap.size());
-        SPARROW_ASSERT_TRUE(std::cmp_greater_equal(values.size(), offset));
-
-        const auto create_buffer = [&values]()
+        const auto size = std::ranges::size(values);
+        SPARROW_ASSERT_TRUE(size == std::ranges::size(bitmap));
+        SPARROW_ASSERT_TRUE(std::cmp_greater_equal(size, offset));
+
+
+        std::vector<array_data::buffer_type> buffers(1);
+        const size_t buffer_size = (size * sizeof(T)) / sizeof(uint8_t);
+        array_data::buffer_type buffer(buffer_size);
+        auto data = buffer.data<T>();
```

Comment:
Should be replaced by `buffer_adaptor`

Reply:
Fyi, I only tried to refactor this, such that it is not using a lambda function`create_buffers` any more.
At the time the lambda function `create_buffers` used to be called, the bitmap-range has already been moved into the array-data struct, therefore I needed to create the buffers before the assignment of `.bitmap = `.

Therefore it did not make sense anymore to do this in a lambda function

Reply:
I would prefer to switch to the buffer_adaptor in a follow-up PR.